app/routes/content.js

```diff
@@ -100,8 +135,24 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, SetupSubmitContro
     },
     actions: {
         error(error, transition) {
-            window.history.replaceState({}, 'preprints', '/preprints/' + transition.params.content.preprint_id);
-            this.intermediateTransitionTo('page-not-found');
+            const slug = transition.params[transition.targetName].preprint_id;
+
+            if (slug.length === 5) {
+                window.location.href = [
+                    window.location.origin,
+                    slug
+                ].join('/');
+            } else {
+                const path = ['', 'preprints'];
+
+                if (this.get('theme.isProvider'))
+                    path.push(this.get('theme.id'));
+
+                path.push(slug);
+
+                window.history.replaceState({}, 'preprints', path.join('/'));
+                this.intermediateTransitionTo('page-not-found');
```

Comment:
This route is reused in routes/provider/content.js, and we have to handle a few different cases with this.

1. `/:guid` redirects to 3, if the provider is not OSF
2. `/preprints/:guid` always redirects to 1
3. `/preprints/:provider/:guid` redirects to 1, if the provider is OSF or to the correct provider if the it's on the wrong provider (e.g. `/preprints/socarxiv/abc12` -> `/preprints/engrxiv/abc12`)

I was running into cases in testing bad subroutes wouldn't be handled correctly, so this was necessary. I really just wanted to make sure we handle it in case a weird edge case comes up where the slug is not 5 characters.